tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php

```diff
@@ -92,6 +94,24 @@ public function dataExpandListParameters()
                 array(1, 2, 3),
                 array(\PDO::PARAM_INT, \PDO::PARAM_INT, \PDO::PARAM_INT)
             ),
+            // DBAL-2685 - Positional: Very simple with one needle, with full type information
+            array(
+                "SELECT * FROM Foo WHERE foo IN (?)",
+                array(array(1, 2, 3)),
+                array('[integer]'),
+                'SELECT * FROM Foo WHERE foo IN (?, ?, ?)',
+                array(1, 2, 3),
+                array('integer', 'integer', 'integer')
+            ),
+            // DBAL-2685 - Positional: Very simple with one needle, with full type information, and a complex type
+            array(
+                "SELECT * FROM Foo WHERE foo IN (?)",
+                array(array($dt, $dt, $dt)),
+                array('[datetime]'),
+                'SELECT * FROM Foo WHERE foo IN (?, ?, ?)',
+                array($dt, $dt, $dt),
+                array('datetime', 'datetime', 'datetime')
```

Comment:
@Ocramius I added tests, I hope my tests are clear: we're trying to bring support for complex types into IN statements, like datetime, or any other custom types